cocos/2d/CCFontFNT.cpp

```diff
@@ -508,20 +510,59 @@ void BMFontConfiguration::parseKerningEntry(const char* line)
     _kerningDictionary[key] = amount;
 }
 
-FontFNT * FontFNT::create(const std::string& fntFilePath, const Vec2& imageOffset /* = Vec2::ZERO */)
+FontFNT * FontFNT::create(const std::string& fntFilePath, const Rect& imageRect, bool imageRotated)
 {
-    BMFontConfiguration *newConf = FNTConfigLoadFile(fntFilePath);
+    const auto newConf = FNTConfigLoadFile(fntFilePath);
     if (!newConf)
         return nullptr;
     
+    const auto tempFont = new FontFNT(newConf, imageRect, imageRotated);
+    
+    tempFont->setFontSize(newConf->_fontSize);
+    if (!tempFont)
+    {
+        return nullptr;
+    }
+    tempFont->autorelease();
+    return tempFont;
+}
+
+FontFNT* FontFNT::create(const std::string& fntFilePath, const std::string& subTextureKey)
+{
+    const auto newConf = FNTConfigLoadFile(fntFilePath);
+    if (!newConf)
+        return nullptr;
+
+    const auto frame = SpriteFrameCache::getInstance()->getSpriteFrameByName(subTextureKey);
+    if (!frame)
+    {
+        return nullptr;
+    }
+    auto tempFont = new FontFNT(newConf, frame->getRectInPixels(), frame->isRotated());
+
+    tempFont->setFontSize(newConf->_fontSize);
+    if (!tempFont)
+    {
+        return nullptr;
+    }
```

Comment:
- should check tempFont before setting font size
- should release newConf before return

Reply:
Yes, that's based on the current code, and wasn't modified by me, but I'll fix it now.